### PR TITLE
Fix tests if ~/.aws/shell/ doesn't exist.

### PR DESCRIPTION
**Pull request by donnemartin**
Fixes issues with test_keys.py and test_config.py.  Creates the directory that will contain the awsshellrc config file if it doesn't yet exist.  Once it exists, continues with copying the template config file to ~/.aws/shell/awsshellrc.

### CODE REVIEW

File: awsshell/config.py [hunk 1]
```diff
@@ -75,9 +75,17 @@ def _copy_template_to_config(self, template_path,
...
+
+        :raises: :class:`OSError <exceptions.OSError>`
         """
         expanded_config_path = os.path.expanduser(config_path)
         if not overwrite and os.path.isfile(config_path):
             return
         else:
+            try:
+                config_path_dir_name = os.path.dirname(config_path)
+                os.makedirs(config_path_dir_name)
```
Comment (donnemartin) [inline awsshell/config.py]: We might want to consolidate the places where we create the ~/.aws/shell/ directory (and sub directories).  I count four separate locations now in the codebase (including this new call) where we are calling `os.makedirs`.

Comment (jamesls) [inline awsshell/config.py]: I agree it would be good to consolidate these calls.

**Comment by jamesls** [comment]
:shipit:  Looks good, thanks!

File: awsshell/config.py (hunk 1 above)
Comment (donnemartin) [inline awsshell/config.py]: Capture this refactor in a new issue: https://github.com/awslabs/aws-shell/issues/65.

**Pull request merged**